src/main/scala/au/com/agiledigital/toolform/command/generate/kubernetes/DeploymentWriter.scala

```diff
@@ -64,8 +68,114 @@ object DeploymentWriter extends KubernetesWriter {
       identity
     }
 
-  private def writeContainer(projectId: String, service: ToolFormService): Result[Unit] = {
-    val imageName   = determineImageName(projectId, service)
+  private def writeVolumeMounts(service: ToolFormService, project: Project): Result[Unit] = {
+    val componentID = service.id
+
+    // look through project topology and find volume claims IDs for current component/resource
+    val relatedVolumeIDs: Seq[String] = project.topology.volumes
+      .getOrElse(Nil)
+      .filter(volume => volume.to.refId == componentID)
+      .map(_.from.refId)
+
+    val volumesWithPaths: Option[Seq[(String, String)]] = Some(relatedVolumeIDs flatMap { volumeId =>
+      findPathsFromVolumeID(project, volumeId).map((volumeId, _))
+    }).filter(_.nonEmpty)
+
+    val maybeVolumeMounts: Option[Result[Unit]] = volumesWithPaths map { list =>
+      list.toList traverse_ {
+        case (volumeId, path) =>
+          val result: Result[Unit] = for {
+            _ <- write(s"- name: ${volumeId}")
+            _ <- write(s"  mountPath: ${path}")
+          } yield ()
+          result
+      }
+    }
+
+    maybeVolumeMounts match {
+      case Some(volumeMounts) =>
+        for {
+          _ <- write("volumeMounts:")
+          _ <- indented(volumeMounts)
+        } yield ()
+      case None => identity
+    }
+  }
+
+  private def findPathsFromVolumeID(project: Project, resourceID: String): List[String] =
+    project.resources.get(resourceID) match {
+      case Some(resource) => {
+        returnVolumePath(resource, resourceID)
+      }
+      case None =>
+        println(s"Warning: '$resourceID' does not exist")
+        Nil
+    }
+
+  private def returnVolumePath(resource: Resource, id: String): List[String] = {
+    val paths = resource.settings match {
+      case Some(settings) => {
+        Try(settings.getStringList("paths")) match {
+          case Success(list) => list
+          case Failure(error) => {
+            println(s"Error: no path specified for resource: $id.")
+            val res: java.util.List[String] = List("error")
```

Comment:
Because that is what the success condition returns. Not specifying that java type returns an `overloaded method value join with alternatives:` error on line 133. Is there a better way of doing this?

Reply:
Could the settings be specified as a case class instead of passing around the settings object? That would mean that the validation of the settings would be done when the config is loaded and then you would have an object which is strongly typed and doesn't need to be wrapped in a try to access it. That would also get rid of the Java types. That's what we do in `ProjectReader`. Would that work?